packages/core/src/lib/lit-core.ts

```diff
@@ -203,8 +204,11 @@ export class LitCore {
   listenForNewEpoch = async (): Promise<void> => {
     const stakingContract = await LitContracts.getStakingContract(this.config.litNetwork as LitNetwork);
 
-    stakingContract.on("StateChanged", async () => {
-      await this.setNewConfig();
+    stakingContract.on("StateChanged", async (state: StakingStates) => {
+      log(`New state detected: "${state}"`);
+      if (state === StakingStates.NextValidatorSetLocked) {
+        await this.setNewConfig();
+      }
```

Comment:
nice, @Adarsh-Kumar28 is there any other state transitions where we would want to re check the validators?